Cookbook/Proposals/DecoupleSnapshotTestsFromUnitTests.md

```diff
@@ -0,0 +1,31 @@
+# Decouple Snapshot tests from unit tests
+* Author(s): Adam Borek
+
+## Introduction
+The purpose of this proposal is to decouple snapshot tests from unit tests to be able to run only unit-tests which takes much less time to execute than snapshot tests.
+
+The goal of this proposal is to make it possible to run only unit-tests **locally** in Xcode. Improving CI run time is not a goal of this proposal.
+
+## Motivation
+Right now, when we trigger all unit-tests (by pressing cmd+U), the snapshot tests are also run.  It takes **14 minutes** to run all unit+snapshot tests which is very long. Those 14 minutes doesn't include the build time 😱.
+
+Unit-tests should run as quickly as possible, so we can have a quick feedback about our changes that break tests.\
+\
+Ideally, it would be nice to be able to run only unit-tests when we don't work on UI. Unit-tests in separation take about ~2% of the execution time.
+
+## Proposed solution
+Decouple snapshot tests from unit test by creating `*SnapshotTests` targets in every framework which has snapshots.
```

Comment:
Side note be I think we could use that occasion to rename our test targets too, i.e. use a dash to separate main name from suffix. That way we'd have `BabylonFoo-Tests` (or `BabylonFoo-UnitTests`, we can bikeshed), `BabylonFoo-SnapshotTests` and `BabylonFoo-UITests` – avoiding the ambiguity of names like `BabylonFooUITests` that are not clear if they are about Unit `Tests` for `BabylonFooUI`, or `UITests` for `BabylonFoo`

Sure it's not the direct topic of this proposal, but since this proposal will add more test targets to the project, I think it's the right time to make sure the name of each test target are clear in intent.

Reply:
Yeah, I agree this ambiguity is irritating. I could also change this :)